core/computing-unit-managing-service/src/main/scala/edu/uci/ics/texera/service/resource/ComputingUnitAccessResource.scala

```diff
@@ -0,0 +1,224 @@
+  }
+
+  /**
+    * Identifies whether the given user is the owner of the given computing unit
+    *
+    * @param cuid computing unit id
+    * @param uid user id
+    * @return boolean value indicating yes/no
+    */
+  def isOwner(cuid: Integer, uid: Integer): Boolean = {
+    val workflowComputingUnitDao = new WorkflowComputingUnitDao(context.configuration())
+    val unit = workflowComputingUnitDao.fetchOneByCuid(cuid)
+    unit != null && unit.getUid.equals(uid)
+  }
+
+  def getPrivilege(cuid: Integer, uid: Integer): PrivilegeEnum = {
+    val computingUnitUserAccessDao = new ComputingUnitUserAccessDao(context.configuration())
+    val accessList = computingUnitUserAccessDao
+      .fetchByUid(uid)
+      .asScala
+      .find(_.getCuid.equals(cuid))
+
+    accessList match {
+      case Some(access) => access.getPrivilege
+      case None         => null
+    }
+  }
+
+  case class AccessEntry(email: String, name: String, privilege: EnumType) {}
+
+}
+
+@Produces(Array(MediaType.APPLICATION_JSON))
+@RolesAllowed(Array("REGULAR", "ADMIN"))
+@Path("/access")
+class ComputingUnitAccessResource {
+  private def ensureSharingIsEnabled(): Unit = {
+    if (!ComputingUnitConfig.sharingComputingUnitEnabled) {
+      throw new ForbiddenException(
+        "The computing unit sharing feature is disabled by the administrator."
+      )
+    }
+  }
+  final private val userDao = new UserDao(context.configuration())
+
+  @GET
+  @Produces(Array(MediaType.APPLICATION_JSON))
+  @Path("/computing-unit/list/{cuid}")
+  def getComputingUnitAccessList(
+      @Auth user: SessionUser,
+      @PathParam("cuid") cuid: Integer
+  ): List[AccessEntry] = {
+    ensureSharingIsEnabled()
+    withTransaction(context) { ctx =>
+      val computingUnitUserAccessDao = new ComputingUnitUserAccessDao(ctx.configuration())
+      computingUnitUserAccessDao
+        .fetchByCuid(cuid)
+        .asScala
+        .map(access => {
+          val user = userDao.fetchOneByUid(access.getUid)
+          AccessEntry(
+            email = user.getEmail,
+            name = user.getName,
+            privilege = access.getPrivilege
+          )
+        })
+        .toList
+    }
+  }
+
+  @PUT
+  @Path("/computing-unit/grant/{cuid}/{email}/{privilege}")
+  def grantAccess(
+      @Auth user: SessionUser,
+      @PathParam("cuid") cuid: Integer,
+      @PathParam("email") email: String,
+      @PathParam("privilege") privilege: PrivilegeEnum
+  ): Unit = {
+    ensureSharingIsEnabled()
+    if (!hasWriteAccess(cuid, user.getUid)) {
+      throw new IllegalArgumentException("User does not have permission to grant access")
+    }
+
+    // TODO: add try except and check how to display error message in the frontend
+    val granteeId = userDao.fetchOneByEmail(email).getUid
+    if (granteeId == null) {
+      throw new IllegalArgumentException("User with the given email does not exist")
+    }
+
+    withTransaction(context) { ctx =>
```

Comment:
as we are doing WRITE access sharing in this PR, can you check if the given privilege is WRITE, if not reject it?